engine/src/org/pentaho/di/trans/Trans.java

```diff
@@ -308,10 +308,10 @@
   private volatile boolean preparing;
 
   /** Whether the transformation is initializing. */
-  private boolean initializing;
+  private volatile boolean initializing;
```

Comment:
Seems we are using three variables to describe the state of execution for the transformation. This seems really weird to me. Why not one AtomicInteger called state where 0 = Initializing, 1 = Running, 2 = Paused, and 3 = Finished. Or, if something can be both initializing and running, then at most 2 variables - executionState and executionPhase - executionState = Running, Paused, Finished and executionPhase = Preparation, Execution, Finalization. Something like that makes more sense to me.